subsys/lorawan/lorawan.c

```diff
@@ -605,9 +616,13 @@ static int lorawan_init(const struct device *dev)
 		return -EINVAL;
 	}
 
+	if (IS_ENABLED(CONFIG_LORAWAN_NVM_SETTINGS)) {
+		settings_subsys_init();
```

Comment:
Can this be abstracted out to a `lorawan_nvm_init()` call.
That would let us remove all references to a particular NVM backend from the main lorawan files.
The `IS_ENABLED` would change to `if(!IS_ENABLED(CONFIG_LORAWAN_NVM_NONE))`